lib/iris/tests/unit/analysis/cartography/test_rotate_grid_vectors.py

```diff
@@ -0,0 +1,148 @@
+import numpy as np
+
+from iris.cube import Cube
+
+from iris.analysis.cartography import rotate_grid_vectors
+from iris.tests.stock import sample_2d_latlons
+
+
+class TestRotateGridVectors(tests.IrisTest):
+    def _check_angles_calculation(self, angles_in_degrees=True,
+                                  nan_angles_mask=None):
+        # Check basic maths on a 2d latlon grid.
+        u_cube = sample_2d_latlons(regional=True, transformed=True)
+        u_cube.units = 'ms-1'
+        u_cube.rename('dx')
+        u_cube.data[...] = 0
+        v_cube = u_cube.copy()
+        v_cube.name('dy')
+
+        # Define 6 different vectors, repeated in each data row.
+        in_vu = np.array([(0, 1), (2, -1), (-1, -1), (-3, 1), (2, 0), (0, 0)])
+        in_angs = np.rad2deg(np.arctan2(in_vu[..., 0], in_vu[..., 1]))
+        in_mags = np.sqrt(np.sum(in_vu * in_vu, axis=1))
+        v_cube.data[...] = in_vu[..., 0]
+        u_cube.data[...] = in_vu[..., 1]
+
+        # Define 5 different test rotation angles, one for each data row.
+        rotation_angles = np.array([0., -45., 135, -140., 90.])
+        ang_cube_data = np.broadcast_to(rotation_angles[:, None],
+                                        u_cube.shape)
+        ang_cube = u_cube.copy()
+        if angles_in_degrees:
+            ang_cube.units = 'degrees'
+        else:
+            ang_cube.units = 'radians'
+            ang_cube_data = np.deg2rad(ang_cube_data)
+        ang_cube.data[:] = ang_cube_data
+
+        if nan_angles_mask is not None:
+            ang_cube.data[nan_angles_mask] = np.nan
+
+        # Rotate all vectors by all the given angles.
+        result = rotate_grid_vectors(u_cube, v_cube, ang_cube)
+        out_u, out_v = [cube.data for cube in result]
+
+        # Check that vector magnitudes were unchanged.
+        out_mags = np.sqrt(out_u * out_u + out_v * out_v)
+        expect_mags = in_mags[None, :]
+        self.assertArrayAllClose(out_mags, expect_mags)
+
+        # Check that vector angles are all as expected.
+        out_angs = np.rad2deg(np.arctan2(out_v, out_u))
+        expect_angs = in_angs[None, :] + rotation_angles[:, None]
+        ang_diffs = out_angs - expect_angs
+        # Fix for null vectors, and +/-360 differences.
+        ang_diffs[np.abs(out_mags) < 0.001] = 0.0
+        ang_diffs = ang_diffs % 360.0
+        # Check that any differences are very small.
+        self.assertArrayAllClose(ang_diffs, 0.0)
+
+        # Check that results are always masked arrays, masked at NaN angles.
+        self.assertTrue(np.ma.isMaskedArray(out_u))
+        self.assertTrue(np.ma.isMaskedArray(out_v))
+        if nan_angles_mask is not None:
+            self.assertArrayEqual(out_u.mask, nan_angles_mask)
+            self.assertArrayEqual(out_v.mask, nan_angles_mask)
+
+    def test_angles_calculation(self):
+        self._check_angles_calculation()
+
+    def test_angles_in_radians(self):
+        self._check_angles_calculation(angles_in_degrees=True)
```

Comment:
This test is called `test_angles_in_radians` but it uses `angles_in_degrees=True`

Reply:
Whoops!  
Should obviously be 'False'. Will fix.